roc_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"math/rand"
+	"reflect"
+	"testing"
+)
+
+func TestROC(t *testing.T) {
+	rnd := rand.New(rand.NewSource(1))
+	cases := []struct {
+		y       []float64
+		c       []bool
+		wantTPR []float64
+		wantFPR []float64
+	}{
+		{
+			y:       []float64{0.0, 1.0, 2.0, 3.0, 4.5, 5.0, 5.5, 6.0, 7.0, 8.0, 9.0, 9.5},
+			c:       []bool{true, true, true, false, true, false, false, false, false, false, false, false},
+			wantTPR: []float64{0.0, 0.25, 0.5, 0.75, 0.75, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0},
+			wantFPR: []float64{0.0, 0.0, 0.0, 0.0, 0.125, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1.0},
+		},
+		{
+			y:       make([]float64, 12),
+			c:       make([]bool, 12),
+			wantTPR: []float64{0.0, 0.25, 0.5, 0.75, 0.75, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0},
+			wantFPR: []float64{0.0, 0.0, 0.0, 0.0, 0.125, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1.0},
+		},
+	}
+	p := rnd.Perm(len(cases[0].y))
+	for i, j := range p {
+		cases[1].y[i] = cases[0].y[j]
+		cases[1].c[i] = cases[0].c[j]
+	}
+	for _, test := range cases {
+		gotTPR, gotFPR := ROC(test.y, test.c)
+		if !reflect.DeepEqual(gotTPR, test.wantTPR) {
+			t.Errorf("unexpected TPR got:%v want:%v", gotTPR, test.wantTPR)
```

Comment:
please replace:
```go
t.Errorf("unexpected TPR got:%v want:%v", gotTPR, test.wantTPR)
```
with:
```go
t.Errorf("test #%d: unexpected TPR got:%v want:%v", i, gotTPR, test.wantTPR)
```
(or something along these lines)

where `i` is coming from:
```go
for i, test := range cases {
```

this way we know which testcase failed.